### ISSUE TITLE
sparklyr error may be due to tmp/hive settings

### ISSUE THREAD

**Issue by dsblr** [rstudio/sparklyr#497]
Hello Sparklyr team,
I would like to draw your attention for the error I am getting. Because this is the error most of the people are getting when they work behind the corporate fire wall.

Error: 

library(sparklyr)
sc <- spark_connect(master = "local",spark_home = "C:\\Users\\user1\\AppData\\Local\\rstudio\\spark\\Cache\\spark-2.0.1-bin-hadoop2.4\\",hadoop_version = "2.4")
t1 <- copy_to(sc,iris)

## no error

but if I change the working directory and execute the same command I am getting the following error.

Error: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
	at org.apache.spark.sql.hive.client.IsolatedClientLoader.createClient(IsolatedClientLoader.scala:258)
	at org.apache.spark.sql.hive.HiveUtils$.newClientForMetadata(HiveUtils.scala:359)
	at org.apache.spark.sql.hive.HiveUtils$.newClientForMetadata(HiveUtils.scala:263)
	at org.apache.spark.sql.hive.HiveSharedState.metadataHive$lzycompute(HiveSharedState.scala:39)
	at org.apache.spark.sql.hive.HiveSharedState.metadataHive(HiveSharedState.scala:38)
	at org.apache.spark.sql.hive.HiveSharedState.externalCatalog$lzycompute(HiveSharedState.scala:46)
	at org.apache.spark.sql.hive.HiveSharedState.externalCatalog(HiveSharedState.scala:45)
	at org.apache.spark.sql.hive.HiveSessionState.catalog$lzycompute(HiveSessionState.scala:50)
	at org.apache.spark.sql.hive.HiveSessionState.catalog(HiveSessionState.scala:48)
	at org.apache.spark.sql.hive.HiveSessionState$$anon$1.<init>(HiveSessionState.scala:63)
	at org.apache.spark.sql.hive.HiveSessionState.analyzer$lzycompute(HiveSessionState.scala:63)
	at org.apache.spark.sql.hive.HiveSessionState.analyzer(HiveSessionState.scala:62)
	at org.apache.spark.sql.execution.QueryExecution.assertAnalyzed(QueryExecution.scala:49)
	at org.apache.spark.sql.Dataset$.ofRows(Dataset.scala:64)
	at org.apache.spark.sql.SparkSession.sql(SparkSession.scala:582)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at sparklyr.Invoke$.invoke(invoke.scala:94)
	at sparklyr.StreamHandler$.handleMethodCall(stream.scala:89)
	at sparklyr.StreamHandler$.read(stream.scala:55)
	at sparklyr.BackendHandler.channelRead0(handler.scala:49)
	at sparklyr.BackendHandler.channelRead0(handler.scala:14)
	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:105)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:308)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:294)
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:308)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:294)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:244)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:308)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:294)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:846)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:131)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:511)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:468)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:382)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:354)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:111)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:137)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.RuntimeException: java.lang.RuntimeException: The root scratch dir: /tmp/hive on HDFS should be writable. Current permissions are: rwx---rwx
	at org.apache.hadoop.hive.ql.session.SessionState.start(SessionState.java:522)
	at org.apache.spark.sql.hive.client.HiveClientImpl.<init>(HiveClientImpl.scala:189)
	... 46 more
Caused by: java.lang.RuntimeException: The root scratch dir: /tmp/hive on HDFS should be writable. Current permissions are: rwx---rwx
	at org.apache.hadoop.hive.ql.session.SessionState.createRootHDFSDir(SessionState.java:612)
	at org.apache.hadoop.hive.ql.session.SessionState.createSessionDirs(SessionState.java:554)
	at org.apache.hadoop.hive.ql.session.SessionState.start(SessionState.java:508)
	... 47 more

I observed the following things:
-- every time I execute any command related to reading file (copy_to or spark_read_csv) its create a metastore_db under the home directory

There is something related to system tmp file. I am not sure what is the issue.

I haven't changed any of the default setting like hive/spark xml file.

Please resolve this decencies and make sparklyr as stable platform for r + spark users

**Comment by javierluraschi** [comment]
@dsblr I was not able to reproduce this, I follow your steps and then ran `setwd("C:/")`, could you share the command you are using to change the working directory?

Also, we've increased logging under windows, so could you kindly try the following?

```
# uninstall sparklyr
devtools::install_github("rstudio/sparklyr")
options("sparklyr.verbose")

# connect as usual and report back the error log
```

**Comment by dsblr** [comment]
setwd("D:\models\pr1\")       ##changing the working directory
sc <- spark_connect(master = "local",spark_home = "C:\Users\user1\AppData\Local\rstudio\spark\Cache\spark-2.0.1-bin-hadoop2.4\",hadoop_version = "2.4")             ##successful
t1 <- copy_to(sc,iris)
after this step I am getting error, which is mentioned above.

**Comment by dsblr** [comment]
@javierluraschi could you please help me to resolve the above error. This solution will solve many sparklyr users problem. Please find some time to look into this.

**Comment by javierluraschi** [comment]
I agree. We need to get to the bottom of the `sun.reflect.NativeConstructorAccessorImpl.newInstance0` Hive configuration issue. I have seen this issue at least 10 times.

We have a way to repro at least one of this instances by: Using Ubuntu 14, install R using `conda install -r r-sparklyr` then attempt `spark_connect`.

Somewhat related to https://github.com/rstudio/sparklyr/issues/653 which was the environment I could track to reproduce this.

**Comment by dsblr** [comment]
@javierluraschi 
Thank for your input. I am sorry I am using Windows machine.

**Comment by javierluraschi** [comment]
Fix commited, please retry with `sparklyr 0.5.6-9002` or newer. You can install using `devtools::install_github("rstudio/sparklyr")`.
